File: src/main/java/net/atos/zac/app/informatieobjecten/converter/RestInformatieobjecttypeConverter.java
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.informatieobjecten.converter;

import static net.atos.client.zgw.util.UriUtilsKt.extractUuid;

import java.net.URI;
import java.util.List;
import java.util.stream.Collectors;

import jakarta.inject.Inject;

import net.atos.client.zgw.ztc.ZtcClientService;
import net.atos.client.zgw.ztc.model.generated.InformatieObjectType;
import net.atos.zac.app.informatieobjecten.model.RestInformatieobjecttype;

public class RestInformatieobjecttypeConverter {

    @Inject
    private ZtcClientService ztcClientService;

    public static RestInformatieobjecttype convert(final InformatieObjectType type) {
        final RestInformatieobjecttype restType = new RestInformatieobjecttype();
        restType.uuid = extractUuid(type.getUrl());
        restType.concept = type.getConcept();
        restType.omschrijving = type.getOmschrijving();
        // we use the uppercase version of this enum in the ZAC backend API
        restType.vertrouwelijkheidaanduiding = type.getVertrouwelijkheidaanduiding().name();
        return restType;
    }

    public static List<RestInformatieobjecttype> convert(final List<InformatieObjectType> informatieobjecttypen) {
        return informatieobjecttypen.stream()
                .map(RestInformatieobjecttypeConverter::convert)
                .collect(Collectors.toList());
    }

    public List<RestInformatieobjecttype> convertFromUris(final List<URI> informatieobjecttypeUris) {
        return informatieobjecttypeUris.stream()
                .map(ztcClientService::readInformatieobjecttype)
                .map(RestInformatieobjecttypeConverter::convert)
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/nl/info/zac/app/informatieobjecten/converter/RestInformatieobjecttypeConverter.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */

package nl.info.zac.app.informatieobjecten.converter

import jakarta.inject.Inject
import nl.info.client.zgw.ztc.ZtcClientService
import nl.info.zac.app.informatieobjecten.model.RestInformatieobjecttype
import nl.info.zac.app.informatieobjecten.model.toRestInformatieobjecttype
import java.net.URI

class RestInformatieobjecttypeConverter @Inject constructor(
    private val ztcClientService: ZtcClientService
) {
    fun convertFromUris(informatieobjecttypeUris: List<URI>): List<RestInformatieobjecttype> =
        informatieobjecttypeUris
            .map { ztcClientService.readInformatieobjecttype(it) }
            .map { it.toRestInformatieobjecttype() }
}
